cmd/mps-control-daemon/mps/daemon.go

```diff
@@ -124,25 +127,51 @@ func (d *Daemon) Start() error {
 	}
 	defer statusFile.Close()
 
+	klog.InfoS("Starting log tailer", "resource", d.rm.Resource())
+	tail := exec.Command("tail", "-n", "+1", "-f", filepath.Join(logDir, "control.log")) //nolint:gosec
+	tail.Stdout = os.Stdout
+	if err := tail.Start(); err != nil {
+		klog.ErrorS(err, "Could not start tail command on control.log; ignoring logs")
+		d.tail = nil
+	} else {
+		d.tail = tail
+	}
+
 	return nil
 }
 
 // Stop ensures that the MPS daemon is quit.
 func (d *Daemon) Stop() error {
-	output, err := d.EchoPipeToControl("quit")
+	_, err := d.EchoPipeToControl("quit")
 	if err != nil {
 		return fmt.Errorf("error sending quit message: %w", err)
 	}
-	klog.InfoS("Shut down MPS", "output", output)
+	klog.InfoS("Stopped MPS control daemon", "resource", d.rm.Resource())
+
+	if d.tail != nil {
+		klog.InfoS("Stopping log tailer", "resource", d.rm.Resource())
+		if err := d.tail.Process.Signal(os.Kill); err != nil {
+			klog.ErrorS(err, "Failed to stop log tailer")
+		}
+
+		err := d.tail.Wait()
+		klog.InfoS("Stopped log tailer", "resource", d.rm.Resource(), "output", err)
+		d.tail = nil
+	}
 
 	if err := d.setComputeMode(computeModeDefault); err != nil {
 		return fmt.Errorf("error setting compute mode %v: %w", computeModeDefault, err)
 	}
 
-	err = os.Remove(d.startedFile())
-	if err != nil && err != os.ErrNotExist {
+	if err := os.Remove(d.startedFile()); err != nil && err != os.ErrNotExist {
 		return fmt.Errorf("failed to remove started file: %w", err)
 	}
+
+	logDir := d.LogDir()
+	if err := os.RemoveAll(logDir); err != nil {
+		klog.ErrorS(err, "Failed to remove pipe directory", "path", logDir)
+	}
```

Comment:
This seems relevant, but unrelated to this PR.

Reply:
Moved to #591 

Note that it manifests here since we're tailing the file from the beginning.

Reply:
Approved other PR